sktime/detection/hmm.py

```diff
@@ -155,6 +155,11 @@ def __init__(
         self.initial_probs = initial_probs
         self.emission_funcs = emission_funcs
         self.transition_prob_mat = transition_prob_mat
+        self.num_states = len(emission_funcs)
+        self.states = list(range(self.num_states))
+        if self.initial_probs is None:
```

Comment:
this needs to use the pattern where we write to `self._initia_probs` (in both cases) instead of mutating the parameter